plugins/speculation-rules/helper.php

```diff
@@ -87,6 +87,12 @@ function plsr_get_speculation_rules() {
 							'href_matches' => $href_exclude_paths,
 						),
 					),
+					// Also exclude rel=nofollow links, as plugins like WooCommerce attribute their add-to-cart links (which should buttons).
```

Comment:
typo here? anyway, maybe remote this part and suggest Woo fix upstream.
```suggestion
					// Also exclude rel=nofollow links, as plugins like WooCommerce attribute their add-to-cart links.
```